### PR TITLE
Prepare crates for publishing

### PR DESCRIPTION
**Pull request by mtreinish**
This commit updates the openqasm3_parser project for publishing crates. Primarily this involves unifying the packaging metadata between all the crates in the workspace. The other big change is all the crates and library names are renamed to have a `oq3_` prefix with the exception of qasm3_bytecode because it already had a descriptive name that was unlikely to conflict with anything.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -3,44 +3,26 @@ members = ["crates/*"]
...
-# text-size.opt-level = 3
-# # This speeds up `cargo xtask dist`.
-# miniz_oxide.opt-level = 3
-
-# [profile.release]
-# incremental = true
-# # Set this to 1 or 2 to get more useful backtraces in debugger.
-# debug = 0
+readme = "README.md"
+keywords = ["QASM", "OpenQASM3", "parser"]
```
Comment (jakelishman) [inline Cargo.toml:12]:
```suggestion
keywords = ["QASM", "OpenQASM 3", "parser"]
```
if we're allowed spaces

File: crates/parser/Cargo.toml
Comment (jakelishman) [inline crates/parser/Cargo.toml:1]: Can we rename the directory as well, just for consistency?

File: crates/qasm3_bytecode/Cargo.toml
Comment (jakelishman) [inline crates/qasm3_bytecode/Cargo.toml:1]: This whole crate can be deleted, I believe.  Certainly it doesn't build with `cargo build` on my machine, and I had to delete it locally to get a working build to write against.

File: crates/semantics/Cargo.toml
Comment (jakelishman) [inline crates/semantics/Cargo.toml:1]: Can we rename the directory for consistency?

**Review by jakelishman — COMMENTED** [review]
Mostly this looks fine to me.  Good to strip the "MIT or " bit from the license, since I don't think we're allowed to have it as part of our OSS strategy.

Do we have a formal requirement to include the same Apache-2.0-ish header on all source files in the distribution package, since `crates.io` involves us distributing the source files?  My approximately understanding was that Apache 2.0 doesn't require it (and explicitly says you shouldn't include the _copyright_ bit for ASF-developed stuff), but I think we might have different rules internally?

Could we unify all the directories for the source files to be `oq3_` prefixed for consistency with their package names?

File: crates/qasm3_bytecode/Cargo.toml
Comment (jakelishman) [reply crates/qasm3_bytecode/Cargo.toml:1]: cc @jlapeyre

File: Cargo.toml (hunk 1 above)
Comment (mtreinish) [reply Cargo.toml:12]: I'd have to check if cargo allows them or not, but I think on crates.io it's normalized to all lower case and with `-` only https://crates.io/keywords

Comment (jakelishman) [reply Cargo.toml:12]: Ah yeah, looks like it.  `openqasm3` is fine by me - it's mostly `OpenQASM3` with the mix of partially using the preferred stylisation but with a lack of space that jars me.

File: crates/parser/Cargo.toml
Comment (mtreinish) [reply crates/parser/Cargo.toml:1]: Fixed in: https://github.com/Qiskit/openqasm3_parser/pull/13/commits/ad75fca8e31c4b4188129a1b880fe3696726dc0a

File: Cargo.toml (hunk 1 above)
Comment (mtreinish) [reply Cargo.toml]: I switched it to openqasm3 in https://github.com/Qiskit/openqasm3_parser/pull/13/commits/ad75fca8e31c4b4188129a1b880fe3696726dc0a

File: crates/semantics/Cargo.toml
Comment (mtreinish) [reply crates/semantics/Cargo.toml:1]: Done in: https://github.com/Qiskit/openqasm3_parser/pull/13/commits/ad75fca8e31c4b4188129a1b880fe3696726dc0a

File: crates/qasm3_bytecode/Cargo.toml
Comment (mtreinish) [reply crates/qasm3_bytecode/Cargo.toml:1]: This was done in #15

**Review by jakelishman — APPROVED** [review]
Thanks, this packaging seems sensible and more uniform to me now.  I'm sure there's references to the old structures in bits and places of comments (the readme, etc), but we can correct those over time.

**Pull request merged**